### ISSUE TITLE
Crash when running SwiftFormat

### ISSUE THREAD

**Issue by 1oo7** [nicklockwood/SwiftFormat#614]
SwiftFormat crashes on line 72 of "ParsingHelpers.swift":
```
        return tokens[start ..< end].reduce(0) { total, token in
```

Because `start` is less than `end`. The error text from the console says:

```
Fatal error: Can't form Range with upperBound < lowerBound
```

Steps to reproduce:

Create a folder "Test". 

Inside this folder create a file "test.swift" with the following contents:

```
struct Test {
    func webView(_: WKWebView, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
        authenticationChallengeProcessor.process(challenge: challenge, completionHandler: completionHandler)
    }
}
```

Also inside folder "Test", create a .swiftformat file with the following contents:

```
# Documentation https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md

# Disabled Rules
--disable andOperator,redundantLetError,redundantReturn,trailingClosures,hoistPatternLet,ranges,trailingCommas,redundantSelf,redundantObjc,fileHeader

# wrapArguments Options
--wraparguments before-first
--wrapcollections before-first
--wrapparameters before-first
--closingparen balanced

# sortedImports Options
--importgrouping testable-bottom

# semicolons Options
--semicolons never

# linebreak Options
--linebreaks lf

# indent Options
--indent 4
--indentcase false
--comments ignore
--ifdef indent
--trimwhitespace always

# maxwidth Options
--maxwidth 80

# numberFormatting Options
--decimalgrouping 3
```

Lastly run swiftformat inside this folder and you will experience the crash.

UPDATE: This bug does not happen if `--maxwidth` is not specified.

**Comment by 1oo7** [comment]
The following line also causes the crash:

```
    func enumerateModelTree(includingSelf: Bool = true, body: (_ model: BaseModel, _ exploreSubtree: inout Bool, _ stop: inout Bool) -> Void) {
```

Evidently, any time there is closure parameter which has multiple input arguments then swiftformat dies...

**Comment by 1oo7** [comment]
Adding the following line after line 1084 of ParsingHelpers.swift seems to fix the bug:

```
                        max(lastIndex, 0) > nextWrapIndex,
```

So the full `if` statement now reads:

```
                    if let nextWrapIndex = indexOfNextWrap(),
                        max(lastIndex, 0) > nextWrapIndex,
                        maxWidth < lineLength(from: max(lastIndex, 0), upTo: nextWrapIndex),
                        !willWrapAtStartOfReturnType(maxWidth: maxWidth) {
                        wrapArgumentsWithoutPartialWrapping()
                        lastIndex = nextWrapIndex
                        return
                    }
```

**Comment by nicklockwood** [comment]
Thanks for reporting, I'll get this fixed asap.

**Comment by 1oo7** [comment]
@nicklockwood I don't know if it's related but, after I implemented the fix by inserting the check `max(lastIndex, 0) > nextWrapIndex,` as shown above, I noticed that we started to see this other issue: 

https://github.com/nicklockwood/SwiftFormat/issues/615

(where wrapping prefers to wrap at the dot accessor over wrapping arguments/parameters).

Since I'm not sure they're related, I made that a separate issue, but at least for my team we feel that the preferences of wrapping at the dot accessor is a dealbreaker to using the `maxwidth` setting.

**Comment by nicklockwood** [comment]
Fixed in 0.44.8

**Comment by 1oo7** [comment]
Thank you @nicklockwood
